packages/main/src/Input.js

```diff
@@ -672,6 +708,34 @@ class Input extends UI5Element {
 		this.focused = true;
 	}
 
+	_handlePageUp(event) {
+		if (!this.focused && this.Suggestions && this.Suggestions.isOpened()) {
+			this.Suggestions.onPageUp(event);
+		} else {
```

Comment:
This code repeats on each of the keystrokes. It should be extracted in a single place and reused:
```
		if (!this.focused && this.Suggestions && this.Suggestions.isOpened()) {
              this.Suggestions.onPageUp(event);
		} else {
```

It could be a check method:
isSuggestionsOpen, for example, so it wouuld become something like this:

```
		if (this.isSuggestionsOpen()) {
              this.Suggestions.onPageUp(event);
		} else {
```

Reply:
I also have to check for the focus state of the Input though. So it would look more like:
```suggestion
		if (!this.focused && this.isSuggestionsOpen()) {
			this.Suggestions.onPageUp(event);
		} else {
```